src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -989,32 +1029,21 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   * @define coll lazy list
   * @define Coll `LazyList`
   */
-@SerialVersionUID(3L)
+@SerialVersionUID(4L)
 object LazyList extends SeqFactory[LazyList] {
-  // Eagerly evaluate cached empty instance
-  private[this] val _empty = newLL(State.Empty).force
-
-  private sealed trait State[+A] extends Serializable {
-    def head: A
-    def tail: LazyList[A]
-  }
+  private object Uninitialized extends Serializable
+  private object MidEvaluation
```

Comment:
this not being `Serializable` is begging for an awful edge case ever with racy serialization

Reply:
I can add `extends Serializable`.

But if a lazy list gets serialized in the middle of `initState` with `_state eq MidEvaluation`, de-serialization creates a LazyList instance that will never be able to initialize (the example below is after making it serializable):

```scala
scala> val ll = 1 #:: { Thread.sleep(1000); 2} #:: LazyList.empty
val ll: scala.collection.immutable.LazyList[Int] = LazyList(<not computed>)

scala> new Thread(() => println(ll.toList)).start(); Thread.sleep(500); val ser = serialize(ll)
val ser: Array[Byte] = Array(...

scala> val lld = deserialize(ser).asInstanceOf[LazyList[Int]]
val lld: LazyList[Int] = LazyList(<not computed>)

scala> lld.toList
java.lang.RuntimeException: LazyList evaluation depends on its own result (self-reference); see docs for more info
  at scala.collection.immutable.LazyList.initState(LazyList.scala:300)
```

Maybe it's better to get a failure during serialization?

For comparison, current 2.13.15 also fails on de-serialization

```scala
scala> val lld = deserialize(ser).asInstanceOf[LazyList[Int]]
java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field scala.collection.immutable.LazyList.lazyState of type scala.Function0 in instance of scala.collection.immutable.LazyList
  at java.base/java.io.ObjectStreamClass$FieldReflector.setObjFieldValues(ObjectStreamClass.java:2096)
```

Reply:
fair point, I think you're right that we should leave it as is

Reply:
👍 added a test